### PR TITLE
Pass bToken only for external/public function calls

### PR DESCRIPTION
**Pull request by jchittoda**
- [x] Debug failures

### CODE REVIEW

File: contracts/bprotocol/avatar/AbsCToken.sol [hunk 1]
```diff
@@ -23,7 +24,7 @@ contract AbsCToken is Cushion {
         return bComptroller.isBToken(bToken);
     }
 
-    function borrowBalanceCurrent(ICToken cToken) public returns (uint256) {
+    function borrowBalanceCurrent(ICToken cToken) public onlyBToken returns (uint256) {
```
Comment (yaronvel) [inline contracts/bprotocol/avatar/AbsCToken.sol:27]: why did you add this modifier?

Comment (jchittoda) [reply contracts/bprotocol/avatar/AbsCToken.sol:27]: on the Avatar side function takes `cToken` as an argument and on the BToken side, it takes `bToken` as an argument. So to avoid the confusion hiding `borrowBalanceCurrent` function on Avatar contract. Only exposing it to the user on BToken side.

File: contracts/bprotocol/avatar/Avatar.sol [hunk 2]
```diff
@@ -64,4 +65,26 @@ contract Avatar is AbsComptroller, AbsCToken {
         uint256 result = super.mint(cToken, mintAmount);
         return result;
     }
+
+    // EMERGENCY FUNCTIONS
+    function resetApprove(IBToken bToken) external onlyAvatarOwner {
```
Comment (yaronvel) [inline contracts/bprotocol/avatar/Avatar.sol:70]: if this is under emergency, maybe just let it reset any token.
it will take an IERC20 as input, and will reset it.

File: contracts/bprotocol/avatar/Avatar.sol [hunk 3]
```diff
@@ -64,4 +65,26 @@ contract Avatar is AbsComptroller, AbsCToken {
...
+        address cToken = bToken.cToken();
+        ICToken(cToken).underlying().safeApprove(cToken, 0);
+    }
+
+    function transferERC20(address token) external onlyAvatarOwner {
```
Comment (yaronvel) [inline contracts/bprotocol/avatar/Avatar.sol:75]: with all trasnfer fee etc, maybe to be safe add `amount` param?
just in case it will not be possible to send everything.

File: contracts/bprotocol/avatar/Avatar.sol [hunk 4]
```diff
@@ -64,4 +65,26 @@ contract Avatar is AbsComptroller, AbsCToken {
...
+        // ensure that token should not be a cToken, this is to protect user Score manipulation
+        require(ICToken(token).isCToken() == false, "Avatar: cToken-not-allowed");
+
+        address owner = registry.ownerOf(address(this));
+        IERC20 erc20 = IERC20(token);
+        // NOTICE: not enclosing in require()
+        erc20.transfer(owner, erc20.balanceOf(address(this)));
+    }
+
+    function transferETH() external onlyAvatarOwner {
```
Comment (yaronvel) [inline contracts/bprotocol/avatar/Avatar.sol:85]: if we add an amount in line 75, then also here.

File: contracts/bprotocol/avatar/AvatarBase.sol [hunk 5]
```diff
@@ -34,6 +34,11 @@ contract AvatarBase is Exponential {
     // Liquidation cToken
     ICToken public liquidationCToken;
 
+    modifier onlyAvatarOwner() {
```
Comment (yaronvel) [inline contracts/bprotocol/avatar/AvatarBase.sol:37]: should we allow it for delegator? no, right?

Comment (jchittoda) [reply contracts/bprotocol/avatar/AvatarBase.sol:37]: no

File: contracts/bprotocol/avatar/Avatar.sol [hunk 6]
```diff
@@ -64,4 +65,25 @@ contract Avatar is AbsComptroller, AbsCToken {
...
+    }
+
+    function transferERC20(address token, uint256 amount) external onlyAvatarOwner {
+        // ensure that token should not be a cToken, this is to protect user Score manipulation
+        require(ICToken(token).isCToken() == false, "Avatar: cToken-not-allowed");
+
+        address owner = registry.ownerOf(address(this));
+        IERC20 erc20 = IERC20(token);
+        // NOTICE: not enclosing in require()
+        erc20.transfer(owner, amount);
```
Comment (yaronvel) [inline contracts/bprotocol/avatar/Avatar.sol:81]: to be consistent, maybe we should not use `safeApprove` in line 71.

**Review by yaronvel — APPROVED** [review]
